doc/source/tokudb/tokudb_variables.rst

```diff
@@ -4,452 +4,1525 @@
+must be flushed or node must be split.
 
-.. variable:: tokudb_create_index_online
+Changing the value of :variable:`tokudb_block_size` only affects subsequently
+created tables and indexes. The value of this variable cannot be changed for an
+existing table/index without a dump and reload.
 
-This variable controls whether indexes created with the ``CREATE INDEX`` command are hot (if enabled), or offline (if disabled). Hot index creation means that the table is available for inserts and queries while the index is being created. Offline index creation means that the table is not available for inserts and queries while the index is being created.
+.. variable:: tokudb_bulk_fetch
 
-.. note:: Hot index creation is slower than offline index creation.
+  :cli: Yes
+  :conf: Yes
+  :scope: Session, Global
+  :dyn: Yes
+  :vartype: Boolean
+  :default: ON
 
-By default, :variable:`tokudb_create_index_online` is enabled.
+This variable determines if our bulk fetch algorithm is used for ``SELECT``
+statements. ``SELECT`` statements include pure ``SELECT ...`` statements, as
+well as ``INSERT INTO table-name ... SELECT ...``, ``CREATE TABLE table-name
+... SELECT ...``, ``REPLACE INTO table-name ... SELECT ...``, ``INSERT IGNORE
+INTO table-name ... SELECT ...``, and ``INSERT INTO table-name ... SELECT ...
+ON DUPLICATE KEY UPDATE``.
 
-.. variable:: tokudb_disable_slow_alter
+.. variable:: tokudb_cache_size
 
-This variable controls whether slow alter tables are allowed. For example, the following command is slow because HCADER does not allow a mixture of column additions, deletions, or expansions:
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Numeric
 
-.. code-block:: mysql
+This variable configures the size in bytes of the |TokuDB| cache table. The
+default cache table size is 1/2 of physical memory. Percona highly recommends
+using the default setting if using buffered I/O, if using direct I/O then
+consider setting this parameter to 80% of available memory.
 
-  ALTER TABLE table
-  ADD COLUMN column_a INT, 
-  DROP COLUMN column_b;
+Consider decreasing :variable:`tokudb_cache_size` if excessive swapping is
+causing performance problems. Swapping may occur when running multiple |MySQL|
+server instances or if other running applications use large amounts of physical
+memory.
 
-By default, :variable:`tokudb_disable_slow_alter` is disabled, and the engine reports back to mysql that this is unsupported resulting in the following output:
+.. variable:: tokudb_cachetable_pool_threads
 
-.. code-block:: bash
+  :version 5.6.26-74.0: Implemented
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Numeric
+  :range: 0 - 1024
+  :default: 0
+
+This variable defines the number of threads for the checkpoint worker thread
+pool. This pool is used to serialize, compress and write nodes cloned during
+checkpoint. It uses old algorithm to set pool size to 4 / num_cpu_threads.
```

Comment:
This variable defines the number of threads for the cachetable worker thread pool. This pool is used to perform node prefetches, and to serialize, compress, and write nodes during cachetable eviction.
The default value of 0 calculates the pool size to be num_cpu_threads * 2.